common/optional.hpp

```diff
@@ -15,6 +15,14 @@
 #ifndef MENDER_COMMON_OPTIONAL_HPP
 #define MENDER_COMMON_OPTIONAL_HPP
 
+// optional-lite is not binary compatible between C++ versions. This is important, since (as per
+// 2023-05) we build cross-platform files with C++11, and platform files with (optionally) a later
+// version. And then mix them.
+//
+// Luckily, it's possible to force optional-lite to stick with C++11 regardless of what the compiler
+// is using. We don't need any later features in this particular library, so just stick to C++11 all
+// the time.
+#define optional_CPLUSPLUS 201103L
```

Comment:
Q: Should this be a config in `CMake`? :thinking:

Reply:
I don't think we need it. This does not need to change as long as we don't need any post-C++11 features, which is entirely dependent on our code, not on user needs.

In about a decade or so, we can remove this entirely, and rely on C++23! 😄